### PR TITLE
nixos/nginx: fix mincore filtering

### PR DESCRIPTION
**Pull request by Mic92**
Mincore is a syscall, not a group.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erikarvstedt** [comment]
I had opened https://github.com/NixOS/nixpkgs/issues/148009 to address this issue.
I'd vote for either removing `mincore` from nginx or adding it to all other services that use negative syscall filters.
To quote myself:
@ajs124, what's the reason for prohibiting mincore? nginx is the only NixOS service where this syscall is explicitly excluded.

cc @Izorkin

**Comment by erikarvstedt** [comment]
This bug is still unfixed.
I'm getting deployment errors on my systems due to `systemd-analyze verify` checks for service units, which fail for `nginx`.

Let's remove mincore for now and maybe re-add it to **all** services that use negative syscall filters in a separate PR.
I'd also be happy with just merging this PR.

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 1]
```diff
@@ -900,7 +900,7 @@ in
         PrivateMounts = true;
         # System Call Filtering
         SystemCallArchitectures = "native";
-        SystemCallFilter = "~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid @mincore";
+        SystemCallFilter = "~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid mincore";
```
Comment (Mic92) [inline nixos/modules/services/web-servers/nginx/default.nix:903]:
```suggestion
        SystemCallFilter = "~@cpu-emulation @debug @keyring @ipc @mount @obsolete @privileged @setuid";
```

**Comment by Mic92** [comment]
dropped and solved merge conflict.

**Review by erikarvstedt — APPROVED** [review]

**Pull request merged**